src/browser/transports/chromium_socket.ts

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2017 OpenFin Inc.
+
+Licensed under OpenFin Commercial License you may not use this file except in compliance with your Commercial License.
+Please contact OpenFin Inc. at sales@openfin.co to obtain a Commercial License.
+*/
+
+import {socketNet} from 'electron';
+import {parse as parseUrl, format as formatUrl, Url} from 'url';
+import * as log from '../log';
+import ofEvents from '../of_events';
+import route from '../../common/route';
+
+import * as nodeNet from 'net';
+
+interface RequestProtocol {
+    port: number;
+    secureProtocol?: string;   // secure version of the protocol
+    chromiumProtocol?: string; // protocol Chromium network layer should use to create connection
+}
+
+const ProtocolMap: { [index: string]: RequestProtocol } = {
+    // tslint:disable-next-line:no-http-string
+    'rtmp:' :  { port: 1935, secureProtocol: 'rtmps:',  chromiumProtocol: 'http:'},
+    // tslint:disable-next-line:no-http-string
+    'rtmps:' : { port: 443,  chromiumProtocol: 'https:'},
+    // tslint:disable-next-line:no-http-string
+    'http:' : { port: 80,  secureProtocol: 'https:'},
+    // tslint:disable-next-line:no-http-string
+    'https:' : { port: 443 }
+};
+
+enum ProxyEventType {
+    Open = 1,       // proxy socket connected
+    Listening,      // proxy socket starts listening to a port
+    ProxyData,      // data from proxy socket
+    ChromiumData,   // data from Chromium socket
+    Closed
+}
+
+interface ProxyEvent {
+    eventType: ProxyEventType;
+    port?: number;
+    payload?: Buffer;
+    clientSocket?: nodeNet.Socket;
+}
+
+interface ProxyAuthEvent {
+    url: string;
+    isProxy: boolean;
+}
+
+const requestMap: { [url: string]: any } = {};  // map of URL to net.request
+const expectedStatusCode = /^[23]/; // 2xx & 3xx status codes are okay
+
+export interface CreateProxyResponse {
+    success: boolean;
+    data?: {port?: number, originalUrl: string}; // port# on localhost
+}
+
+export interface CreateProxyRequest {
+    url: string;            // URL to proxy requests to
+    callback: (result: CreateProxyResponse) => void;
+    errorCallback: (err: any) => void;
+}
+
+export interface AuthProxyRequest {
+    url: string; // URL for the original CreateProxyRequest
+    username: string;
+    password: string;
+}
+
+export function createChromiumSocket(req: CreateProxyRequest): void {
+    log.writeToLog(1, `createChromiumSocket: ${JSON.stringify(req)}`, true);
+    createFullProxySocket(req);
```

Comment:
why not put `createFullProxySocket`'s logic in here?

Reply:
I will add more code here to support another feature